addons/mail/models/mail_channel.py

```diff
@@ -746,6 +746,41 @@ def _channel_message_notifications(self, message, message_format=False):
     #   - when a message is posted on a channel (to the channel, using _notify() method)
     # ------------------------------------------------------------
 
+    def set_message_pin(self, message_id, pinned):
+        """ (Un)pin a message on the channel and send a notification to the
+        members.
+        :param message_id: id of the message to be pinned.
+        :param pinned: whether the message should be pinned or unpinned.
+        """
+        self.ensure_one()
+        message_to_update = self.message_ids.filtered(
+            lambda msg: msg.id == message_id and bool(msg.pinned_at) != pinned
+        )
+        if not message_to_update:
+            return
+        message_to_update.write({'pinned_at': fields.datetime.now() if pinned else False})
+        self.env['bus.bus']._sendone(self, 'mail.record/insert', {
+            'Message': {
+                'id': message_id,
+                'pinned_at': fields.Datetime.to_string(message_to_update.pinned_at),
+            }
+        })
```

Comment:
Do we still want to support/introduce more of the old syntax?

Reply:
If the purpose is to just insert a record, it alright. Better keep consistency until we change all of them at once